From 89303c44ed3c65e106f7e2e7711c112867226695 Mon Sep 17 00:00:00 2001
From: DonRehan <30264386+donRehan@users.noreply.github.com>
Date: Tue, 4 Jun 2024 04:33:48 +0300
Subject: [PATCH] Add a dash above current active window in dwm bar

Give user ability to modify its position , width and text position.
---
 config.def.h | 3 +++
 dwm.c        | 4 +++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 9efa774..2ec5f52 100644
--- a/config.def.h
+++ b/config.def.h
@@ -2,6 +2,9 @@
 
 /* appearance */
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
+static const unsigned int brdsh_w  = 2;        /* width of the app bar dash */
+static const unsigned int brdsh_ypos  = 18;        /* y-position of the dash */
+static const unsigned int text_ypos  = 1;        /* y-position of text */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
diff --git a/dwm.c b/dwm.c
index f1d86b2..3c9b293 100644
--- a/dwm.c
+++ b/dwm.c
@@ -736,8 +736,10 @@ drawbar(Monitor *m)
 
 	if ((w = m->ww - tw - x) > bh) {
 		if (m->sel) {
+			drw_setscheme(drw, scheme[SchemeNorm]);
+			drw_text(drw, x, text_ypos, w, bh, lrpad / 2, m->sel->name, 0);
 			drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
-			drw_text(drw, x, 0, w, bh, lrpad / 2, m->sel->name, 0);
+			drw_rect(drw, x , bh - brdsh_ypos , w , brdsh_w , 1, 1);
 			if (m->sel->isfloating)
 				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
 		} else {
-- 
2.34.1

